docs/src/index.md

```diff
@@ -1,407 +1,258 @@
+
+Non-standard types can be provided, like `Dec64` from the DecFP.jl package, but must support the `Base.tryparse(T, str)` function for parsing a value from a string. This allows, for example, easily defining a custom type, like `struct Float64Array; values::Vector{Float64}; end`, as long as a corresponding `Base.tryparse` definition is defined, like `Base.tryparse(::Type{Float64Array}, str) = Float64Array(map(x -> parse(Float64, x), split(str, ';')))`, where a single cell in the data input is like `1.23;4.56;7.89`.
+
+Note that the default [stringtype](@ref stringtype) can be overridden by providing a column's type manually, like `CSV.File(source; types=Dict(1 => String), stringtype=PosLenString)`, where the first column will be parsed as a `String`, while any other string columns will have the `PosLenString` type.
+
+### Examples
+  * [Matrix-like Data](@ref matrix_example)
+  * [Providing types](@ref types_example)
+
+## [`typemap`](@id typemap)
+
+A `Dict{Type, Type}` argument that allows replacing a non-`String` standard type with another type when a column's type is auto-detected. Most commonly, this would be used to force all numeric columns to be `Float64`, like `typemap=Dict(Int64 => Float64)`, which would cause any columns detected as `Int64` to be parsed as `Float64` instead. Another common case would be wanting all columns of a specific type to be parsed as strings instead, like `typemap=Dict(Date => String)`, which will cause any columns detected as `Date` to be parsed as `String` instead.
+
+### Examples
+  * [Typemap](@ref typemap_example)
+
+## [`pool`](@id pool)
+
+Argument that controls whether columns will be returned as `PooledArray`s. Can be provided as a `Bool`, `Float64`, vector of `Bool` or `Float64`, or dict mapping column number/name to `Bool` or `Float64`. As a `Bool`, controls absolutely whether a column will be pooled or not; if passed as a single `Bool` argument like `pool=true`, then all string columns will be pooled, regardless of cardinality. When passed as a `Float64`, the value should be between `0.0` and `1.0` indicating the threshold under which the % of unique values found in the column will result in the column being pooled. For example, if `pool=0.1`, then all string columns with a unique value % less than 10% will be returned as `PooledArray`, while other string columns will be normal string vectors. As mentioned, when the `pool` argument is a single `Bool` or `Float64`, only string columns will be considered for pooling. When a vector or dict is provided, the pooling for any column can be provided as a `Bool` or `Float64`. Similar to the [types](@ref types) argument, providing a vector to `pool` should have an element for each column in the data input, while a dict argument can map column number/name to `Bool` or `Float64` for specific columns. Unspecified columns will not be pooled when the argument is a dict.
```

Comment:
If I understand correctly if single value for `pool` is used it is applied to string columns only and otherwise if it is a vector or dict to any columns. Maybe highlight it more?

Also in this description I do not see what is the default (maybe I missed it, but I cannot find it)

Reply:
Ah yes, I considered this section not-quite-done. I think it'd be useful for you, @nalimilan, and I (and anyone else interested) to discuss what the default pooling behavior will be. On current `main` branch, the default pool value is `0.25`, and it applies the same to single and multithreading. The multithreaded case has also been updated w/ the same behavior as single threaded: i.e. we'll parse the entire file, and post-parsing, we'll check the cardinality of _potentially_ pooled columns to see if they pass the threshold and then convert to regular array or pooled array as appropriate. The use-case I'm worried about given the current state is if you have a large file that uses multiple threads to parse, and you have a very high cardinality string column (like a string id). In that case, we would do roughly double the work parsing that id column, only to discard the pooled values at the end.

An alternative I've considered is making the default `pool=false`. So by default, no surprises or potential slowdowns. If you want pooled columns, you'd have to pass a pool value or `pool=true` explicitly.

Reply:
I am OK with both `pool=0.25.` and `pool=false` as a default. I think the problems users have with pooled columns should be resolved (like the `map` issue) on our side.

Reply:
Would it be too weird to have `pool=false` for multithreaded parsing and `pool=0.25` for single-threaded? Would we run into too many issues where the difference of array would be weird?